src/loaders/agent.js

```diff
@@ -45,11 +47,19 @@ export class Agent {
                     this.features[f.featureName] = new f(this.agentIdentifier, this.sharedAggregator)
                 }
             })
-            gosNREUMInitializedAgents(this.agentIdentifier, this.features, 'features')
+            gosNREUMInitializedAgents(this.agentIdentifier, this.features, NR_FEATURES_REF_NAME);
         } catch (err) {
-            warn(`Failed to initialize instrument classes`, err)
-            // unwrap window apis to their originals
-            // remove agent if initialized to free resources ?
+            warn(`Failed to initialize all enabled instrument classes (agent aborted) -`, err)
+            for (const featName in this.features) { // this.features hold only features that have been instantiated
+                this.features[featName].abortHandler?.();
+            }
+
+            const newrelic = gosNREUM();
+            delete newrelic.initializedAgents[this.agentIdentifier]?.['api'];   // prevent further calls to agent-specific APIs (see "configure.js")
+            delete newrelic.initializedAgents[this.agentIdentifier]?.[NR_FEATURES_REF_NAME];    // GC mem used internally by features
+            delete this.sharedAggregator;
+            // Keep the initialized agent object with its configs for troubleshooting purposes.
+            delete newrelic.ee?.get(this.agentIdentifier);  // clear our events storage
```

Comment:
should we call ee[namespace].abort() here?

Reply:
EE's Abort right now 1) is "global", and 2) sets flag and points ref to a new object. Whereas here we're outright deleting the entire ee object itself, so there's a slight distinction.